### ISSUE TITLE
"block-opening-brace-newline-before" chokes on a comment before an opening brace

### ISSUE THREAD

**Issue by PhiLhoSoft** [stylelint/stylelint#1148]
Reproduced with Stylelint 6.2.1
Initially reported in https://github.com/AtomLinter/linter-stylelint/issues/176 as the plugin crashes on this "error".

Scss file: sample.scss
```
body // boom!
{
	font-family: $mo-font;
	font-weight: normal;
	color: $font-color;
	line-height: normal;
}
```
Rules:
```
rules:
  block-opening-brace-newline-before: "always-multi-line"
```
Command:

    stylelint --syntax scss sample.scss

Result:
```
sample.scss
 2:14  ×  Expected newline before "{" of a multi-line block   block-opening-brace-newline-before
```
The error is not reported if I remove the line comment. Still there if I use a block comment instead.

**Comment by davidtheclark** [comment]
Makes sense. Allowing for end-of-line comments has had to be done on a rule-by-rule basis, so I guess this particular situation just hasn't come up for anyone before. Will mark as a bug.

**Comment by jeddy3** [comment]
This seems like a good time to go through the existing implementations and see if it's possible to abstract out some of this into a utility:

- https://github.com/stylelint/stylelint/blob/master/src/rules/declaration-block-semicolon-newline-after/index.js#L41-L50
- https://github.com/stylelint/stylelint/blob/master/src/rules/block-opening-brace-newline-after/index.js#L44-L54
- https://github.com/stylelint/stylelint/blob/master/src/rules/at-rule-semicolon-newline-after/index.js#L28-L42
- https://github.com/stylelint/stylelint/blob/master/src/rules/at-rule-semicolon-newline-after/index.js#L28-L42

@PhiLhoSoft Feel free to get involved if you're itching for this issue to be resolved :)

**Comment by davidtheclark** [comment]
I created a utility for those other rules, but then realized that the problem with this particular bug is not the  same thing. I think it uncovered a bug in the way PostCSS is generating `rule.raws.between`: https://github.com/postcss/postcss/pull/798
